UI/lib/elements.html

```diff
@@ -432,7 +432,7 @@
 
 <?lsmb BLOCK tooltip;
   IF element_data && element_data.id.defined() && element_data.msg.defined();
-     props="connectId:'" _ element_data.id _ "'";
+     props="connectId:'" _ element_data.id.replace('_','-') _ "'";
```

Comment:
this doesn't make sense. Why would an id have an underscore that needs replacement with a hyphen?

It feels very much like this is papering over another problem: If the 'id' is explicitly set, we definitely shouldn't be changing it.

Reply:
I'm not changing the id of the main item, Dojo take cares of normalizing underscores to hyphens, so we need to point the tooltip to the normalized widgetid.

Reply:
It does? I know LedgerSMB "normalizes" items with a "name=" with an underscore to have an "id=" with a hyphen. Isn't your problem caused by the underscore-to-hyphen-in-LedgerSMB-html-generator code instead of by Dojo?

Reply:
The problem wasn't Dojo but us, trying to get nicer CSS in our Template elements.

Reply:
Can you be more verbose? This remark is too coarse. I have no idea what you're referring to.

Reply:
I added a comment in elements.html where we "normalize" id but affect tooltips, for which we compensate in tooltip generation.

Reply:
Actually, no, we don't. The code you commented doesn't apply to all ids so we should not compensate all cases: inputs which themselves provide an explicit "id" value don't get this transcoding (only those ids which are derived from "name" attributes)

Reply:
Fixed, we now normalize the button and the related tooltip at the same point. I removed the Tooltip patch.